mesonbuild/modules/pkgconfig.py

```diff
@@ -192,7 +192,11 @@ def _fn(xs, libs=False):
             for x in xs:
                 # Don't de-dup unknown strings to avoid messing up arguments like:
                 # ['-framework', 'CoreAudio', '-framework', 'CoreMedia']
-                if x not in result or (libs and (isinstance(x, str) and not x.endswith(('-l', '-L')))):
+                known_flags = ['-pthread']
+                cannot_dedup = libs and isinstance(x, str) and \
+                               not x.startswith(('-l', '-L')) and \
+                               x not in known_flags
```

Comment:
__[Flake8]__ <sub>posted by Sider</sub>

[E127] continuation line over-indented for visual indent